### PR TITLE
Returns more information (name, desc, status, severity)

### LINKED ISSUE #19: Client should be able to know checks and associate errors to these checks
For the client to be able to list the checks that are being executed, and to associate errors to checks:

- a command (for example `doa list-checks`) should return the list of checks supported by the tool[1]
- the `analyze` command should associate each error with a check[2]

[1] list of checks
```
$ doa list-checks -o json
[
  { "name": "user", "description": "Container User" },
  { "name": "port", "description": "Container Exposed port" },
  [...]
]
```

[2] analyze result
```
$ doa analyze -f Dockerfile -o json
[
  { "check": "user", "message": "USER directive set to root in parent image parent1" },
  { "check": "user", "message": "USER directive set to root" },
  { "check": "port", "message": "port 80 exposed at line 28 could be wrong. TCP/IP port numbers below 1024 are privileged port numbers" },
  [...]
]
```

### PR DESCRIPTION
**Pull request by feloy**
Fixes #19 

I'm not sure about the severity of each result

### CODE REVIEW

File: pkg/command/expose.go [hunk 1]
```diff
@@ -22,19 +22,29 @@ import (
...
+	errs := []Result{}
 	str := node.Value
 	index := strings.IndexByte(node.Value, '/')
 	if index >= 0 {
 		str = node.Value[0:index]
 	}
 	port, err := strconv.Atoi(str)
 	if err != nil {
-		errs = append(errs, err)
+		errs = append(errs, Result{
```
Comment (jeffmaury) [inline pkg/command/expose.go:34]:
```suggestion
		results = append(results, Result{
```

File: pkg/command/expose.go [hunk 2]
```diff
@@ -22,19 +22,29 @@ import (
...
+			Name:        "Wrong port value",
+			Status:      StatusFailed,
+			Severity:    SeverityCritical,
+			Description: err.Error(),
+		})
 	}
 	if port < 1024 {
-		errs = append(errs, fmt.Errorf(`port %d exposed %s could be wrong. TCP/IP port numbers below 1024 are privileged port numbers`, port, GenerateErrorLocation(source, line)))
+		errs = append(errs, Result{
```
Comment (jeffmaury) [inline pkg/command/expose.go:42]:
```suggestion
		results = append(results, Result{
```

File: pkg/command/from.go [hunk 3]
```diff
@@ -22,15 +23,20 @@ type From struct {
 
 const SCRATCH_IMAGE_NAME = "scratch"
 
-func (f From) Analyze(node *parser.Node, source utils.Source, line Line) []error {
+func (f From) Analyze(node *parser.Node, source utils.Source, line Line) []Result {
 	if node.Value == SCRATCH_IMAGE_NAME {
 		return nil
 	}
-	errs := []error{}
+	errs := []Result{}
```
Comment (jeffmaury) [inline pkg/command/from.go:30]:
```suggestion
	results := []Result{}
```

File: pkg/command/from.go [hunk 4]
```diff
@@ -22,15 +23,20 @@ type From struct {
...
 	decompiledNode, err := decompiler.Decompile(node.Value)
 	if err != nil {
 		// unable to decompile base image
-		errs = append(errs, fmt.Errorf("unable to analyze the base image %s", node.Value))
+		errs = append(errs, Result{
```
Comment (jeffmaury) [inline pkg/command/from.go:34]:
```suggestion
		results = append(results, Result{
```

File: pkg/command/run.go [hunk 5]
```diff
@@ -21,26 +21,26 @@ import (
 
 type Run struct{}
 
-func (r Run) Analyze(node *parser.Node, source utils.Source, line Line) []error {
-	errs := []error{}
+func (r Run) Analyze(node *parser.Node, source utils.Source, line Line) []Result {
+	errs := []Result{}
```
Comment (jeffmaury) [inline pkg/command/run.go:25]:
```suggestion
	results := []Result{}
```

File: pkg/command/run.go [hunk 6]
```diff
@@ -21,26 +21,26 @@ import (
...
 
 	// let's split the run command by &&. E.g chmod 070 /app && chmod 070 /app/routes && chmod 070 /app/bin
 	splittedCommands := strings.Split(node.Value, "&&")
 	for _, command := range splittedCommands {
 		if r.isChmodCommand(command) {
 			err := r.analyzeChmodCommand(command, source, line)
 			if err != nil {
-				errs = append(errs, err)
+				errs = append(errs, *err)
```
Comment (jeffmaury) [inline pkg/command/run.go:33]:
```suggestion
				results = append(results, *err)
```

File: pkg/command/run.go [hunk 7]
```diff
@@ -21,26 +21,26 @@ import (
...
 			}
 		} else if r.isChownCommand(command) {
 			err := r.analyzeChownCommand(command, source, line)
 			if err != nil {
-				errs = append(errs, err)
+				errs = append(errs, *err)
```
Comment (jeffmaury) [inline pkg/command/run.go:38]:
```suggestion
				results = append(results, *err)
```

**Review by jeffmaury — APPROVED** [review]
LGTM but would like @lstocchi to review the check severities

**Review by lstocchi — APPROVED** [review]
LGTM
I don't have a strong opinion about the severity checks but they look reasonable. Nice job 👍

**Pull request merged**